pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -169,8 +174,91 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 		#[pallet::call_index(0)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
-		pub fn create_pool(_origin: OriginFor<T>) -> DispatchResult {
-			todo!()
+		pub fn create_pool(
+			origin: OriginFor<T>,
+			curve: CurveInput<CurveParameterInputOf<T>>,
+			currencies: BoundedVec<TokenMetaOf<T>, T::MaxCurrencies>,
+			state: PoolStatus<Locks>,
```

Comment:
Is it possible to create a pool with a `Destroying` state?

Reply:
If not, then this should not be a parameter and the state enum should have a default state, kinda like we do for the switch pallet.

Reply:
that's the way I had set it up but SOMEBODY though they could just mess with my beautiful beautiful code !!!!11111

Reply:
I introduced a default value: [341ddf8](https://github.com/KILTprotocol/kilt-node/pull/768/commits/341ddf8a33ef12d53dede9f9adca2821ac4737a3)